packages/git/src/browser/git-quick-open-service.ts

```diff
@@ -38,9 +40,66 @@ export class GitQuickOpenService {
         @inject(Git) protected readonly git: Git,
         @inject(GitRepositoryProvider) protected readonly repositoryProvider: GitRepositoryProvider,
         @inject(QuickOpenService) protected readonly quickOpenService: QuickOpenService,
-        @inject(MessageService) protected readonly messageService: MessageService
+        @inject(MessageService) protected readonly messageService: MessageService,
+        @inject(WorkspaceService) protected readonly workspaceService: WorkspaceService,
+        @inject(FileSystem) protected readonly fileSystem: FileSystem
     ) { }
 
+    async clone(url?: string, folder?: string, branch?: string): Promise<string | undefined> {
+        if (!folder) {
+            const roots = await this.workspaceService.roots;
+            folder = roots[0].uri;
+        }
+
+        if (url) {
+            return this.git
+                .clone(url,
+                    {
+                        localUri: await this.buildDefaultProjectPath(folder, url),
+                        branch: branch
+                    })
+                .then(repo => repo.localUri);
+        }
+
+        const __folder = folder;
+        const __this = this;
+        const cloneRepoModel: QuickOpenModel = {
+            onType(lookFor: string, acceptor: (items: QuickOpenItem[]) => void): void {
+                const dynamicItems: QuickOpenItem[] = [];
+                const suffix = "Press 'Enter' to confirm or 'Escape' to cancel.";
+                if (lookFor === undefined || lookFor.length === 0) {
+                    dynamicItems.push(new SingleStringInputOpenItem(`Please provide a Git repository URI. ${suffix}`, () => { }, () => false));
```

Comment:
I agree with @kittaakos , URL should be used or `address` or `link`

Reply:
@kittaakos @benoitf what about `Git repository location` ?

Reply:
fine for me